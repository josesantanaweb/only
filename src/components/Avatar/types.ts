export interface AvatarProps {
  /** Sets size avatar */
  size?: "small" | "medium" | "large";
  /** Sets img avatar */
  img?: string;
  /** Sets name avatar */
  name?: string;
}
