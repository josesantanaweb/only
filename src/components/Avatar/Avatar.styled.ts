/* eslint-disable consistent-return */
import styled, { css } from "styled-components";
import { AvatarProps } from "./types";

const wrapperModifiers = {
  small: () => css`
    width: 45px;
    height: 45px;
  `,
  medium: () => css`
    width: 60px;
    height: 60px;
  `,
  large: () => css`
    width: 90px;
    height: 90px;
  `,
};

const AvatarStyled = styled.div<AvatarProps>(
  ({ theme, size }) => css`
    display: inline-block;
    border-radius: 100%;
    display: flex;
    align-items: center;
    color: ${theme?.colors.white};
    justify-content: center;
    overflow: hidden;
    // FIXME: Eliminar luego de agregar el component text
    h3 {
      font-weight: 400;
    }
    ${!!size && wrapperModifiers[size]};
  `
);

export default AvatarStyled;
