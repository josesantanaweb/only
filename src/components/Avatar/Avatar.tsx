import React, { FC } from "react";
import AvatarStyled from "./Avatar.styled";
import { AvatarProps } from "./types";

const Avatar: FC<AvatarProps> = ({
  size = "medium",
  img,
  name = "Default",
}) => {
  const nameFirtsLetter = name?.substr(0, 1);
  return (
    <AvatarStyled size={size}>
      {img ? <img src={img} alt="avatar" /> : <h3>{nameFirtsLetter}</h3>}
    </AvatarStyled>
  );
};

export default Avatar;
