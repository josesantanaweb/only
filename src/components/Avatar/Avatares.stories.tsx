import React from "react";

import { ComponentStory, ComponentMeta } from "@storybook/react";

import Avatar from "./Avatar";

export default {
  title: "Avatar",
  component: Avatar,
  argTypes: {
    name: {
      type: "string",
    },
    size: {
      control: {
        type: "select",
        options: ["small", "medium", "large"],
      },
    },
  },
} as ComponentMeta<typeof Avatar>;

const Template: ComponentStory<typeof Avatar> = (args) => <Avatar {...args} />;

export const Default = Template.bind({});
Default.args = {
  name: "Jose",
  size: "medium",
};

export const Imagen = Template.bind({});
Imagen.args = {
  img: "https://i.pravatar.cc/150?img=12",
  size: "large",
};
