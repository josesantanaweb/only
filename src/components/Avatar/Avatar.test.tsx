import React from "react";
import { render } from "../../test";
import Avatar from "./Avatar";

describe("Avatar", () => {
  it("renders the avatar correctly", () => {
    const { container } = render(<Avatar />);
    expect(container).toMatchSnapshot();
  });

  it("should render with img prop given", () => {
    const { container } = render(
      <Avatar size="medium" img="https://i.pravatar.cc/150?img=3" />
    );
    expect(container).toMatchSnapshot();
  });
});
