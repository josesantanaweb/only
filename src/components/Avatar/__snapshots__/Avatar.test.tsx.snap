// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Avatar renders the avatar correctly 1`] = `
<div>
  <div
    class="sc-bdnxRM hxYBZh"
  >
    <h3>
      D
    </h3>
  </div>
</div>
`;

exports[`Avatar should render with all the props given 1`] = `
<div>
  <div
    class="sc-bdnxRM hxYBZh"
  >
    <img
      alt="avatar"
      src="https://i.pravatar.cc/150?img=3"
    />
  </div>
</div>
`;
